test/aqua.jl

```diff
@@ -0,0 +1,30 @@
+using Test
+using TaylorSeries
+using Aqua
+
+@testset "Aqua tests (performance)" begin
+    # This tests that we don't accidentally run into
+    # https://github.com/JuliaLang/julia/issues/29393
+    # Aqua.test_unbound_args(TaylorSeries)
+    ua = Aqua.detect_unbound_args_recursively(TaylorSeries)
+    @test length(ua) == 0
+
+    # See: https://github.com/SciML/OrdinaryDiffEq.jl/issues/1750
+    # Test that we're not introducing method ambiguities across deps
+    ambs = Aqua.detect_ambiguities(TaylorSeries; recursive = true)
+    pkg_match(pkgname, pkdir::Nothing) = false
+    pkg_match(pkgname, pkdir::AbstractString) = occursin(pkgname, pkdir)
+    filter!(x -> pkg_match("TaylorSeries", pkgdir(last(x).module)), ambs)
+    if VERSION ≠ v"1.10.0-DEV"
+        @test length(ambs) == 0
+    end
```

Comment:
Julia nightly is still failing. I think the problem with this `if` block is that it still returns `true`:
```julia
julia> v"1.10.0-DEV.1471" ≠ v"1.10.0-DEV"
true
```
so that test is still evaluated in Julia nightly. For the time being, I think we should replace it by `if VERSION < v"1.10.0-DEV"`.

Hopefully it works...